hyperspy/io_plugins/attolight.py

```diff
@@ -0,0 +1,351 @@
+        metadata.set_item("Signal.background", bkg)
+        return
+    except:
+        return
+
+
+def file_reader(filename, attolight_acquisition_system='cambridge_uk_attolight',
+                *args, **kwds):
+    """Loads data into CLSEMSpectrum lumispy object.
+    Reads the HYPCard.bin file, containing the hyperspectral CL data.
+    Reads the metadata files contained in the same folder as HYPCard.bin.
+        - MicroscopeStatus.txt to load all the metadata values
+        - Background file contained in the same folder
+        - SEM image (taken simultaneously) with the hyperspectral map.
+
+    Parameters
+    ----------
+    filename : str, None
+        The HYPCard.bin filepath for the file to be loaded, created by
+        the AttoLight software.
+        If None, a pop-up window will be loaded.
+    attolight_acquisition_system : str
+        Specify which acquisition system the HYPCard was taken with, from the
+        attolight_systems dictionary file. By default, it assumes it is
+        the Cambridge Attolight SEM system.
+    lazy : bool
+        If True the file will be opened lazily, i.e. without actually reading
+        the data from the disk until required. Allows datasets much larger than
+        available memory to be loaded.
+
+    Returns
+    -------
+    s : Signal
+        A CLSEMSpectrum lumispy object containing the loaded data with the metadata,
+        the background (if available) and the respective SEM image.
+
+    NOTE
+    ----
+
+    The lumispy package must be installed for these files to run.
+    https://github.com/LumiSpy/lumispy
+
+    """
+
+    # Get folder name
+    hypcard_folder = os.path.split(os.path.abspath(filename))[0]
+
+    # Get folder name (which is the experiment name)
+    name = os.path.basename(hypcard_folder)
+
+    if attolight_acquisition_system == 'cambridge_attolight' and len(name) > 37:
+        # CAUTION: Specifically delimeted by Attolight default naming system
+            name = name[:-37]
+
+    # Create metadata dictionary
+    meta = DictionaryTreeBrowser({
+        'General': {'title': name},
+        "Signal": {'signal_type': 'CL_SEM',
+                   'background': None},
+    })
+
+    # Import metadata
+    metadata_file_name \
+        = attolight_systems[attolight_acquisition_system]['metadata_file_name']
+
+    binning, nx, ny, FOV, grating, central_wavelength_nm, channels, amplification, readout_rate, \
+    exposure_time_ccd_s, dwell_time_scan_s, beam_acc_voltage_kv, gun_lens_amps, obj_lens_amps, aperture_um, \
+    chamber_pressure_torr, real_magnification = \
+        _get_metadata(hypcard_folder, metadata_file_name, attolight_acquisition_system)
+
+    # Add all parameters as metadata
+    _store_metadata(meta, hypcard_folder, metadata_file_name, attolight_acquisition_system)
+
+    # Load data
+    with open(filename, 'rb') as f:
+        data = np.fromfile(f, dtype=[('bar', '<i4')], count=channels * nx * ny)
```

Comment:
The data always have 3 dimensions?

Reply:
Yes. Always 2 navigation dimensions and 1 signal dimension.

Reply:
Is there no way to record and save linescans or single spectra in the Attolight software?